lib/gold_miner/author_config.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module GoldMiner
+  class AuthorConfig
+    DEFAULT_AUTHOR_LINK = "#to-do"
+    DEFAULT_CONFIG_PATH = "lib/config/author_config.yml"
+
+    def self.default
+      YAML
+        .load_file(DEFAULT_CONFIG_PATH)
+        .transform_values { |author| author["link"] }
+        .then { |links| new(links) }
+    end
+
+    def initialize(author_config = {})
+      @author_config = author_config
+    end
+
+    def link_for(slack_username)
+      @author_config.fetch(slack_username, DEFAULT_AUTHOR_LINK)
+    end
```

Comment:
minor: Just noticed some "name disparity": `AuthorConfig` cannot be initialized with the contents of `author_config.yml` because it actually expects "author links" in the initialization.

If we plan to support more config fields in the future, we could load the hash from the YAML file as is, not transforming it, and update the `link_for` method. Something along the lines:

```rb
    def self.default
      YAML
        .load_file(DEFAULT_CONFIG_PATH)
        .then { |author_config| new(author_config) }
    end

    def initialize(author_config = {})
      @author_config = author_config
    end

    def link_for(slack_username)
      @author_config.dig(slack_username, "link") || DEFAULT_AUTHOR_LINK
    end
```

Otherwise, we could rename the file and class like `author_config.yml` --> `author_links.yml` and `AuthorConfig` --> `AuthorLinks`

Reply:
> Alternatively, we could rename the file and class like author_config.yml --> author_links.yml (and update the content) and AuthorConfig --> AuthorLinks

That's how it started, but I figured we could add more fields (like name) later, so I changed the class to `AuthorConfig`.

> AuthorConfig cannot be initialized with the contents of author_config.yml because it actually expects "author links" in the initialization.

You're absolutely right! This is reminiscent of when it was `AuthorLinks` instead. TY for catching this.